LeakCanary crashes instant apps due to creating notifications/not handling SecurityException

When an app is ran as instant app, it is apparently not allowed to create notifications.

When LeakCanary is installed, it obviously creates many notifications and the app will crash when this happens.

Another instant app related bug: #1196 

```
2019-03-10 17:20:07.419 1925-1956/de.chagemann.regexcrossword E/AndroidRuntime: FATAL EXCEPTION: LeakCanary-Heap-Dump
    Process: de.chagemann.regexcrossword, PID: 1925
    java.lang.SecurityException: Instant app de.chagemann.regexcrossword cannot create notifications
        at android.os.Parcel.createException(Parcel.java:1950)
        at android.os.Parcel.readException(Parcel.java:1918)
        at android.os.Parcel.readException(Parcel.java:1868)
        at android.app.INotificationManager$Stub$Proxy.enqueueNotificationWithTag(INotificationManager.java:1619)
        at android.app.NotificationManager.notifyAsUser(NotificationManager.java:405)
        at android.app.NotificationManager.notify(NotificationManager.java:370)
        at android.app.NotificationManager.notify(NotificationManager.java:346)
        at com.squareup.leakcanary.internal.LeakCanaryInternals.showNotification(LeakCanaryInternals.java:159)
        at com.squareup.leakcanary.DefaultLeakDirectoryProvider.requestWritePermissionNotification(DefaultLeakDirectoryProvider.java:172)
        at com.squareup.leakcanary.DefaultLeakDirectoryProvider.listFiles(DefaultLeakDirectoryProvider.java:72)
        at com.squareup.leakcanary.DefaultLeakDirectoryProvider.newHeapDumpFile(DefaultLeakDirectoryProvider.java:89)
        at com.squareup.leakcanary.AndroidHeapDumper.dumpHeap(AndroidHeapDumper.java:71)
        at com.squareup.leakcanary.RefWatcher.ensureGone(RefWatcher.java:142)
        at com.squareup.leakcanary.RefWatcher$1.run(RefWatcher.java:117)
        at com.squareup.leakcanary.AndroidWatchExecutor$3.run(AndroidWatchExecutor.java:80)
        at android.os.Handler.handleCallback(Handler.java:873)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:193)
        at android.os.HandlerThread.run(HandlerThread.java:65)
     Caused by: android.os.RemoteException: Remote stack trace:
        at com.android.server.notification.NotificationManagerService.checkDisqualifyingFeatures(NotificationManagerService.java:4489)
        at com.android.server.notification.NotificationManagerService.enqueueNotificationInternal(NotificationManagerService.java:4406)
        at com.android.server.notification.NotificationManagerService$11.enqueueNotificationWithTag(NotificationManagerService.java:2295)
        at android.app.INotificationManager$Stub.onTransact(INotificationManager.java:126)
        at android.os.Binder.execTransact(Binder.java:752)
2019-03-10 17:20:07.443 1363-3644/? W/ActivityManager:   Force finishing activity de.chagemann.regexcrossword/.SelectCategoryActivity
```

--- a/leakcanary-android/src/main/java/leakcanary/internal/LeakCanaryUtils.kt
+++ b/leakcanary-android/src/main/java/leakcanary/internal/LeakCanaryUtils.kt
@@ -150,7 +150,11 @@ internal object LeakCanaryUtils {
       buildNotification(context, builder)
     val notificationManager =
       context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
-    notificationManager.notify(notificationId, notification)
+    try {
+      notificationManager.notify(notificationId, notification)
+    } catch (ignored: SecurityException) {
+      // https://github.com/square/leakcanary/issues/1197
+    }
   }
 
   fun buildNotification(

